### PR TITLE
Repurpose requirements files to no longer frequently update

### LINKED ISSUE #465: Repurpose requirements files

### PR DESCRIPTION
**Pull request by tp832944**
closes #465 

### PR Type

- Build related changes
- CI related changes
- Documentation content changes

### Description

Repurpose requirements files to no longer frequently update, letting versions go stale.

Disable dependabot for pip except for security updates. This is configured through GitHub.

GitHub Actions now use latest versions of dependencies that pip can find. Still test docs on Linux with pinned versions and unit tests on Windows, MacOS and Linux with pinned versions.

Shrink requirements files into one of each type covering all supported Python versions.

Update documentation to match this change of purpose.

### How Has This Been Tested?

 Test A: GitHub Actions pass

### Does this PR introduce a breaking change?

Yes. Names of requirements files have changed.

### Checklist before requesting a review
- [x] I have made sure that my PR is not a duplicate.
- [x] My code follows the style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -34,6 +34,7 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
+# These match the suffices on the various requirements-*.txt files
```
Comment (tc85324) [inline pyproject.toml:37]: Should this be `suffixes` instead of `suffices`?

**Review by tc85324 — CHANGES_REQUESTED** [review]
Generally looks good to me! I think the requirements line in [.readthedocs.yaml](https://github.com/gchq/coreax/blob/main/.readthedocs.yaml) needs updating though. Also, what is the plan for when only the pinned build/test fails?

File: pyproject.toml (hunk 1 above)
Comment (tp832944) [reply pyproject.toml]: You learn something new every day!

**Comment by tp832944** [comment]
> Generally looks good to me! I think the requirements line in [.readthedocs.yaml](https://github.com/gchq/coreax/blob/main/.readthedocs.yaml) needs updating though. Also, what is the plan for when only the pinned build/test fails?

readthedocs fixed - I had missed that.

If the pinned build fails, fixes to the versions and/or code should be made manually to the bot's pull request to make them work.

File: .readthedocs.yaml [hunk 2]
```diff
@@ -15,4 +15,7 @@ sphinx:
 
 python:
    install:
-    - requirements: requirements-doc-3.11.txt
+    - method: pip
+      path: .
+      extra_requirements:
+        - doc
```
Comment (tc85324) [inline .readthedocs.yaml:21]: Do we want to point to the pinned version here instead. This is based on [RTD recommendations](https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html).

Comment (tp832944) [reply .readthedocs.yaml]: Much as I struggle to be sold on reproducible builds, I can't argue against following official guidelines.

**Review by tc85324 — APPROVED** [review]

**Pull request merged**